### ISSUE TITLE
failing to configure in stackage nightly

### ISSUE THREAD

**Issue by juhp** [fpco/executable-hash#2]
executable-hash-0.2.0.{2,3} seem to fail to configure in Stackage Nightly

```
> /tmp/stackage-build12/executable-hash-0.2.0.3$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.3 --htmldir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.3 --haddockdir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.3 --flags=

System/Executable/Hash/Internal.hs:10:1: error:
    Failed to load interface for ‘Crypto.Hash.SHA1’
    It is a member of the hidden package ‘cryptohash-0.11.9’.
```

**Comment by snoyberg** [comment]
I think the problem is that, in the `hide` section, we've hidden cryptohash instead of cryptohash-sha1. If we switch that around and force both packages to recompile, it should work.

**Comment by snoyberg** [comment]
Ugh, it's worse than this, because cyrptohash conflicts with cryptonite too. This would work as a short-term solution, but longer-term we need custom-setup support in stackage-curator.

**Comment by juhp** [comment]
Sorry, hide both?

**Comment by snoyberg** [comment]
Short term: hide cryptohash-sha1, expose cryptohash.

Longer term: add custom-setup support to avoid this problem entirely, and then probably hide cryptohash as well.

**Comment by juhp** [comment]
Hopefully fixed by fpco/stackage#2181, thanks!

**Comment by decentral1se** [comment]
On the latest nightly build, I see:

```
WARNING: Package executable-hash marked as build success, but not registered (pending: 421, failures: 12)
executable-hash: /tmp/stackage-build12/executable-hash-0.2.0.4/dist/doc/html/executable-hash: openDirStream: does not exist (No such file or directory) (pending: 421, failures: 12)
```

With `nightly/executable-hash-0.2.0.4/build.out ` reporting:
```
> /tmp/stackage-build12$ stack unpack executable-hash-0.2.0.4
Unpacked executable-hash-0.2.0.4 to /tmp/stackage-build12/executable-hash-0.2.0.4/
> /tmp/stackage-build12/executable-hash-0.2.0.4$ runghc -clear-package-db -global-package-db -package-db=/var/stackage/work/builds/nightly/pkgdb Setup configure --package-db=clear --package-db=global --package-db=/var/stackage/work/builds/nightly/pkgdb --libdir=/var/stackage/work/builds/nightly/lib --bindir=/var/stackage/work/builds/nightly/bin --datadir=/var/stackage/work/builds/nightly/share --libexecdir=/var/stackage/work/builds/nightly/libexec --sysconfdir=/var/stackage/work/builds/nightly/etc --docdir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.4 --htmldir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.4 --haddockdir=/var/stackage/work/builds/nightly/doc/executable-hash-0.2.0.4 --flags= -hide-all-packages --package=Cabal --package=base --package=bytestring --package=cryptohash --package=directory --package=file-embed --package=filepath --package=template-haskell
Prepare to build the package.

Usage: Setup configure [FLAGS]

Configure how the package is built by setting package (and other) flags.

The configuration affects several other commands, including build, test,
bench, run, repl.
... [cut output for brevity]
```

**Comment by snoyberg** [comment]
Bug on my part in stackage-curator, just pushed a commit.

**Comment by juhp** [comment]
Hopefully fixed by fpco/stackage#2181, thanks!
